src/WingetCreateCLI/Commands/NewCommand.cs

```diff
@@ -168,6 +164,23 @@ public override async Task<bool> Execute()
                 TelemetryManager.Log.WriteEvent(commandEvent);
             }
         }
+
+        private static void PromptPropertiesAndDisplayManifests(Manifests manifests)
+        {
+            PromptRequiredProperties(manifests.VersionManifest);
+            PromptRequiredProperties(manifests.InstallerManifest, manifests.VersionManifest);
+            PromptRequiredProperties(manifests.DefaultLocaleManifest, manifests.VersionManifest);
+
+            Console.WriteLine();
+            if (Prompt.Confirm(Resources.AddOptionalFields_Message))
+            {
+                PromptOptionalProperties(manifests.DefaultLocaleManifest);
+                PromptOptionalProperties(manifests.InstallerManifest);

```

Comment:
I'm not sure if this is what we really want. This will prompt for optional properties on the InstallerManifest _root_, which includes properties which affect all installers (like MinimumOSVersion and InstallerLocale), but doesn't prompt for optional properties on the Installer nodes themselves.

Also, prompting only for string properties seems overly restrictive. For example, why can't the user set the Scope property here? That seems like one users would commonly want to do, since that's one of the more common optional properties.

Additionally, I don't think we should (ever?) prompt for some properties, like PackageFamilyName, which we can get from the MSIX, and ProductCode, which we can get from the MSI.

(@denelon / @KevinLaMS)

Reply:
So far this is the list of optional string fields that will be prompted. 
- Channel
- InstallerLocale
- MinimumOSVersion
- PackageFamilyName
- ProductCode
- PublisherUrl
- PublisherSupportUrl
- PrivacyUrl
- Author
- PackageUrl
- LicenseUrl
- Copyright
- CopyrightUrl
- Description
- Moniker

This is only the initial PR to allow users the ability to modify the most basic fields. Adding more complicated fields that take the form of enum, list<T>, object, will require more time to get those working properly. If we are trying to get the full functionality of optional fields working, then we will have to wait until next release. I do agree that some of the properties you pointed out should not be asked if we can parse them out.

Reply:
An alternative would be to just prompt the values from the DefaultLocaleManifest and handle the optional fields from the installer manifest in a separate PR.

Reply:
I like that better as an initial implementation, that's much simpler to wrap my mind around. If we go that route, I'd also include Tags since that's another popular field. PromptProperty() already supports string[] properties, so you shouldn't have to do anything special for it, just make sure that the example mentions that they should be comma-delimited

Reply:
Channel isn't implemented yet. I'd keep that one out.
I know Tags is an array so that may be worth holding off on, but that would be a high priority for moderators.

Reply:
Thanks for the feedback! I've decided to only prompt optional fields from the defaultLocaleManifest. I've also spent some time working on adding Tags. I believe it works now and should work as expected.